sql: panic: expected *DBytes, found *tree.DString

Occured on `cyan` running b6b6d0a
```
panic: expected *DBytes, found *tree.DString

goroutine 7963243 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.MustBeDBytes(0x2fa4c80, 0xc4290bc850, 0xc4290bc850, 0x2fa4c80)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/datum.go:1262 +0xe9
github.com/cockroachdb/cockroach/pkg/server.(*adminServer).DataDistribution(0xc42004fe40, 0x2fa22c0, 0xc42b9c0570, 0x4789ab0, 0xc42004fe40, 0xc425a74a08, 0xc42e660440)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/admin.go:1542 +0xa8d
github.com/cockroachdb/cockroach/pkg/server/serverpb._Admin_DataDistribution_Handler.func1(0x2fa22c0, 0xc42b9c0570, 0x28a13a0, 0x4789ab0, 0x28a13a0, 0x4789ab0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/admin.pb.go:1955 +0x86
github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x2fa22c0, 0xc42b9c0570, 0x28a13a0, 0x4789ab0, 0xc42b14d8a0, 0xc42b14d8c0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:48 +0xa33
github.com/cockroachdb/cockroach/pkg/rpc.NewServerWithInterceptor.func1(0x2fa22c0, 0xc42b9c0570, 0x28a13a0, 0x4789ab0, 0xc42b14d8a0, 0xc42b14d8c0, 0x26f2ce0, 0x4789ab0, 0x29f31e0, 0xc42a32e400)
        /go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:168 +0xb9
github.com/cockroachdb/cockroach/pkg/server/serverpb._Admin_DataDistribution_Handler(0x2a18c00, 0xc42004fe40, 0x2fa22c0, 0xc42b9c0570, 0xc42e9bf110, 0xc4203d1fc0, 0x0, 0x0, 0xc42a32e400, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/admin.pb.go:1957 +0x167
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc420c16540, 0x2fc0ba0, 0xc4206f6000, 0xc42a32e400, 0xc420938f00, 0x41947e8, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1011 +0x4fc
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleStream(0xc420c16540, 0x2fc0ba0, 0xc4206f6000, 0xc42a32e400, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1249 +0x1318
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4202ae060, 0xc420c16540, 0x2fc0ba0, 0xc4206f6000, 0xc42a32e400)
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:680 +0x9f
created by github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:678 +0xa1

```

--- a/pkg/server/admin.go
+++ b/pkg/server/admin.go
@@ -1539,8 +1539,8 @@ func (s *adminServer) DataDistribution(
 	for _, row := range rows2 {
 		zcID := int64(tree.MustBeDInt(row[0]))
 		zcCliSpecifier := string(tree.MustBeDString(row[1]))
-		zcYaml := tree.MustBeDBytes(row[2])
-		zcBytes := tree.MustBeDBytes(row[3])
+		zcYaml := tree.MustBeDString(row[2])
+		zcBytes := tree.MustBeDBytes(row[4])
 		var zcProto config.ZoneConfig
 		if err := protoutil.Unmarshal([]byte(zcBytes), &zcProto); err != nil {
 			return nil, s.serverError(err)

--- a/pkg/server/admin_test.go
+++ b/pkg/server/admin_test.go
@@ -1323,8 +1323,6 @@ func TestAdminAPIFullRangeLog(t *testing.T) {
 func TestAdminAPIDataDistribution(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	t.Skip("#24802")
-
 	testCluster := serverutils.StartTestCluster(t, 3, base.TestClusterArgs{})
 	defer testCluster.Stopper().Stop(context.Background())
 
@@ -1381,7 +1379,7 @@ func TestAdminAPIDataDistribution(t *testing.T) {
 	// Wait for the new tables' ranges to be created and replicated.
 	testutils.SucceedsSoon(t, func() error {
 		var resp serverpb.DataDistributionResponse
-		if err := getAdminJSONProto(firstServer, "replica_matrix", &resp); err != nil {
+		if err := getAdminJSONProto(firstServer, "data_distribution", &resp); err != nil {
 			t.Fatal(err)
 		}
 
@@ -1393,50 +1391,6 @@ func TestAdminAPIDataDistribution(t *testing.T) {
 		// Don't test anything about the zone configs for now; just verify that something is there.
 		if len(resp.ZoneConfigs) == 0 {
 			return fmt.Errorf("no zone configs returned")
-		}
-
-		return nil
-	})
-
-	// Add a zone config.
-	sqlDB.Exec(t, `ALTER TABLE roachblog.posts CONFIGURE ZONE USING num_replicas = 1`)
-
-	expectedNewZoneConfigID := int64(51)
-	sqlDB.CheckQueryResults(
-		t,
-		`SELECT zone_id
-		FROM [SHOW ALL ZONE CONFIGURATIONS]
-		WHERE cli_specifier = 'roachblog.posts'`,
-		[][]string{
-			{fmt.Sprintf("%d", expectedNewZoneConfigID)},
-		},
-	)
-
-	// Verify that we see the zone config and its effects.
-	testutils.SucceedsSoon(t, func() error {
-		var resp serverpb.DataDistributionResponse
-		if err := getAdminJSONProto(firstServer, "replica_matrix", &resp); err != nil {
-			t.Fatal(err)
-		}
-
-		postsTableInfo := resp.DatabaseInfo["roachblog"].TableInfo["posts"]
-
-		// Verify that the TableInfo for roachblog.posts points at the new zone config.
-		if postsTableInfo.ZoneConfigId != expectedNewZoneConfigID {
-			t.Fatalf(
-				"expected roachblog.posts to have zone config id %d; had %d",
-				expectedNewZoneConfigID, postsTableInfo.ZoneConfigId,
-			)
-		}
-
-		// Verify that the num_replicas setting has taken effect.
-		numPostsReplicas := int64(0)
-		for _, count := range postsTableInfo.ReplicaCountByNodeId {
-			numPostsReplicas += count
-		}
-
-		if numPostsReplicas != 1 {
-			return fmt.Errorf("expected 1 replica; got %d", numPostsReplicas)
 		}
 
 		return nil

